end-to-end-test/local/specs/core/comparison-alterations-tab.spec.js

```diff
@@ -0,0 +1,339 @@
+            );
+        });
+
+        it('unchecks the master CNA checkbox when one CNA type is unchecked', function() {
+            clickAlterationTypeCheckBox('Deletion');
+            assert(
+                !$('input[data-test=CheckCopynumberAlterations]').isSelected(),
+                'unchecks master Copy Number Alterations checkbox'
+            );
+        });
+
+        it('checks all CNA types using the master CNA checkbox', function() {
+            clickAlterationTypeCheckBox('Deletion');
+            clickAlterationTypeCheckBox('Copy Number Alterations');
+            assert(
+                cnaCheckboxRefs.every(ref =>
+                    $('input[data-test=' + ref + ']').isSelected()
+                ),
+                'checks all CNA checkboxes'
+            );
+        });
+
+        it('checks the master CNA checkbox when all CNA types are checked', function() {
+            clickAlterationTypeCheckBox('Deletion');
+            assert(
+                !$('input[data-test=CheckCopynumberAlterations]').isSelected()
+            );
+            clickAlterationTypeCheckBox('Deletion');
+            assert(
+                $('input[data-test=CheckCopynumberAlterations]').isSelected()
+            );
+        });
+
+        // -+=+ INFRAME MUTATION +=+-
+        it('unchecks all inframe mutation types using the inframe mutation master checkbox', function() {
+            clickAlterationTypeCheckBox('Inframe');
+            assert(
+                inframeCheckboxRefs.every(
+                    ref => !$('input[data-test=' + ref + ']').isSelected()
+                ),
+                'unchecks all inframe mutation checkboxes'
+            );
+        });
+
+        it('unchecks the master inframe mutation checkbox when one inframe mutation type is unchecked', function() {
+            clickAlterationTypeCheckBox('Inframe Insertion');
+            assert(
+                !$('input[data-test=InFrame]').isSelected(),
+                'unchecks master inframe mutation checkbox'
+            );
+        });
+
+        it('checks all inframe mutation types using the master inframe mutation checkbox', function() {
+            clickAlterationTypeCheckBox('Inframe Insertion');
+            clickAlterationTypeCheckBox('Inframe');
+            assert(
+                inframeCheckboxRefs.every(ref =>
+                    $('input[data-test=' + ref + ']').isSelected()
+                ),
+                'checks all inframe mutation checkboxes'
+            );
+        });
+
+        it('checks the master inframe mutation checkbox when all inframe mutation types are checked', function() {
+            clickAlterationTypeCheckBox('Inframe Insertion');
+            assert(!$('input[data-test=InFrame]').isSelected());
+            clickAlterationTypeCheckBox('Inframe Insertion');
+            assert($('input[data-test=InFrame]').isSelected());
+        });
+
+        // -+=+ FRAMESHIFT MUTATION +=+-
```

Comment:
I love your ornate headers :D

Reply:
It was around Christmas that I did this :)